driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -291,23 +346,32 @@ private void sendCommandMessage(final CommandMessage message,
                 compressedBsonOutput.close();
             }
         }
+        responseTo = message.getId();
     }
 
-    private <T> T receiveCommandMessageResponse(final CommandMessage message, final Decoder<T> decoder,
-                                                final CommandEventSender commandEventSender, final SessionContext sessionContext) {
-        ResponseBuffers responseBuffers = receiveMessage(message.getId());
-        try {
+    private <T> T receiveCommandMessageResponse(final Decoder<T> decoder,
+                                                final CommandEventSender commandEventSender, final SessionContext sessionContext,
+                                                final int additionalTimeout) {
+        try (ResponseBuffers responseBuffers = receiveMessageWithAdditionalTimeout(additionalTimeout)) {
             updateSessionContext(sessionContext, responseBuffers);
             if (!isCommandOk(responseBuffers)) {
-                throw getCommandFailureException(responseBuffers.getResponseDocument(message.getId(), new BsonDocumentCodec()),
+                throw getCommandFailureException(responseBuffers.getResponseDocument(responseTo, new BsonDocumentCodec()),
                         description.getServerAddress());
             }
 
-            commandEventSender.sendSucceededEvent(responseBuffers);
+            if (commandEventSender != null) {
+                commandEventSender.sendSucceededEvent(responseBuffers);
+            }
+
+            T commandResult = getCommandResult(decoder, responseBuffers, responseTo);
 
-            return getCommandResult(decoder, responseBuffers, message.getId());
-        } finally {
-            responseBuffers.close();
+            hasMoreToCome = responseBuffers.getReplyHeader().hasMoreToCome();
```

Comment:
This is the main part of the new support the the moreToCome bit

Reply:
Ack